File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第9章：几何形状的检测和拟合/9.1/boxPoints_OpenCV3.cpp
#include<opencv2/core.hpp>
#include<opencv2/highgui.hpp>
#include<opencv2/imgproc.hpp>
using namespace cv;
#include<vector>
#include<iostream>
using namespace std;
int main(int argc, char*argv[])
{
	//构造旋转矩形
	RotatedRect rRect(Point2f(200,200),Point2f(90,150),-60);
	//计算旋转矩形的四个顶点，存储为一个 4 行 2 列的单通道 float 类型的 Mat
	Mat vertices;
	boxPoints(rRect, vertices);
	//打印四个顶点
	cout << vertices << endl;
	//在黑色画板上画出该旋转矩形
	Mat img=Mat::zeros(Size(400, 400), CV_8UC1);
	for (int i = 0; i < 4; i++)
	{
		//相邻的点
		Point p1 = vertices.row(i);
		int j = (i + 1) % 4;
		Point p2 = vertices.row(j);
		//画出直线
		line(img, p1, p2, Scalar(255), 3);
	}
	//显示旋转矩形
	imshow("旋转矩形", img);
	waitKey(0);
	return 0;
}

File: OpenCV演算法精解：基於Python與C++/code/unzip/OpenCV算法精解-代码/第9章：几何形状的检测和拟合/9.1/boxPoints_OpenCV3.py
# -*- coding: utf-8 -*-
import cv2
import numpy as np
#主函数
if __name__ =="__main__":
    #旋转矩形
    vertices = cv2.boxPoints(((200, 200), (90, 150), -60.0))
    #四个顶点
    print vertices.dtype#打印数据类型
    print vertices#打印四个顶点
    #根据四个顶点在黑色画板上画出该矩形
    img=np.zeros((400,400),np.uint8)
    for i in xrange(4):
        #相邻的点
        p1 = vertices[i,:]
        j = (i+1)%4
        p2 = vertices[j,:]
        #画出直线
        cv2.line(img,(p1[0],p1[1]),(p2[0],p2[1]),255,2)
    cv2.imshow("img",img)
    cv2.imwrite("boxPoints.jpg",img)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

